src/Deserializers/LegacyStatementDeserializer.php

```diff
@@ -18,29 +18,38 @@
  */
 class LegacyStatementDeserializer implements Deserializer {
 
-	private $claimDeserializer;
+	/**
+	 * @var Deserializer
+	 */
+	private $snakDeserializer;
+
+	/**
+	 * @var Deserializer
+	 */
 	private $snakListDeserializer;
 
+	/**
+	 * @var array
+	 */
 	private $serialization;
 
-	public function __construct(
-		Deserializer $claimDeserializer,
-		Deserializer $snakListDeserializer
-	) {
-		$this->claimDeserializer = $claimDeserializer;
+	public function __construct( Deserializer $snakDeserializer, Deserializer $snakListDeserializer ) {
+		$this->snakDeserializer = $snakDeserializer;
 		$this->snakListDeserializer = $snakListDeserializer;
 	}
 
 	/**
-	 * @param mixed $serialization
-	 *
+	 * @param array $serialization
 	 * @return Statement
 	 * @throws DeserializationException
 	 */
 	public function deserialize( $serialization ) {
```

Comment:
It's good to have a single level of abstraction per method.

Granted, this was not perfect to begin with, since the array keys and associated messages are details from the requirement list perspective. (I think it'd not be a good trade-off to fix that though.)

So I'm not very happy with this inlining.